pkg/vm/context.go

```diff
@@ -209,3 +209,20 @@ func (c *Context) atBreakPoint() bool {
 func (c *Context) String() string {
 	return "execution context"
 }
+
+
+// GetContextScriptHash returns script hash of the invocation stack element
+// number n.
+func GetContextScriptHash(v *VM, n int) util.Uint160 {
+	ctxIface := v.Istack().Peek(n).Value()
+	ctx := ctxIface.(*Context)
+	return ctx.ScriptHash()
+}
+
+// PushContextScriptHash pushes to evaluation stack the script hash of the
+// invocation stack element number n.
+func PushContextScriptHash(v *VM, n int) error {
```

Comment:
`func (v *VM) PushContextScriptHash(n int) error`? I can't see any reason for it being a standalone function rather than a VM method (and `GetContextScriptHash` too).

Reply:
Yes, it is nicer so.